pkg/remote/aws/sqs_queue_policy_supplier.go

```diff
@@ -0,0 +1,66 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/aws/aws-sdk-go/service/sqs/sqsiface"
+	"github.com/cloudskiff/driftctl/pkg/remote/deserializer"
+	remoteerror "github.com/cloudskiff/driftctl/pkg/remote/error"
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	"github.com/cloudskiff/driftctl/pkg/resource/aws"
+	awsdeserializer "github.com/cloudskiff/driftctl/pkg/resource/aws/deserializer"
+	"github.com/cloudskiff/driftctl/pkg/terraform"
+	"github.com/sirupsen/logrus"
+	"github.com/zclconf/go-cty/cty"
+)
+
+type SqsQueuePolicySupplier struct {
+	reader       terraform.ResourceReader
+	deserializer deserializer.CTYDeserializer
+	client       sqsiface.SQSAPI
+	runner       *terraform.ParallelResourceReader
+}
+
+func NewSqsQueuePolicySupplier(provider *TerraformProvider) *SqsQueuePolicySupplier {
+	return &SqsQueuePolicySupplier{
+		provider,
+		awsdeserializer.NewSqsQueuePolicyDeserializer(),
+		sqs.New(provider.session),
+		terraform.NewParallelResourceReader(provider.Runner().SubRunner()),
+	}
+}
+
+func (s SqsQueuePolicySupplier) Resources() ([]resource.Resource, error) {
+	queues, err := listSqsQueues(s.client)
```

Comment:
Maybe we could start to make every of our new supplier independent from each other ? WDYT @moadibfr @wbeuil

Reply:
would be a good idea to start implementing mutualisation for list in a correct way yep.

Reply:
Yep let's start that, I agree. Would it be what you expect if I make it independant by creating methods that list sqs queues for each of the 2 resources ? Or you have another way in your mind ?

Reply:
> creating methods that list sqs queues for each of the 2 resources

I had that in mind

Reply:
I had more in mind a custom client with shared function and that compose the base client so it could be use as a replacement. You can look in sqs_topic pr for an example

Reply:
Maybe this could be a dedicated PR which can come in the future ? It's a pure performance related topic imho it could be done outsided theses new resources PR

Reply:
Let's do this in 2 steps:

The first one with dedicated methods in this PR.

Then, let's talk about Martin's client in his PR (SNS topic) and open issues to switch all suppliers to a newer and more efficient architecture.